build.go

```diff
@@ -889,3 +903,39 @@ func parseDigestFromImageID(id imgutil.Identifier) string {
 	digest = strings.TrimPrefix(digest, "sha256:")
 	return fmt.Sprintf("sha256:%s", digest)
 }
+
+func createInlineBuildpack(bp project.Buildpack, stackID string) (string, error) {
+	pathToInlineBuilpack, err := ioutil.TempDir("", "inline-cnb")
+	if err != nil {
+		return pathToInlineBuilpack, err
+	}
+
+	if err = createBuildpackTOML(pathToInlineBuilpack, bp.ID, "0.0.0", bp.Script.API, []dist.Stack{{ID: stackID}}, nil); err != nil {
+		return pathToInlineBuilpack, err
+	}
+
+	shell := bp.Script.Shell
+	if shell == "" {
+		shell = "/bin/sh"
+	}
+
+	binBuild := fmt.Sprintf(`#!%s
```

Comment:
#! looks like it could cause problems on windows machines. And might be useful to have an integration test with one of these inline buildpacks just to show things do work in both envs.

Reply:
do you mean windows as the host machine, or windows containers (or both)?

Reply:
Windows Containers

Reply:
Should we have a different default shell for windows containers?

Reply:
> Should we have a different default shell for windows containers?

Technically, Windows doesn't need an explicit shell chosen, nor a shebang line, just needs appropriate file suffixes (`detect.bat` / `build.bat`) since `cmd.Start()` will automatically call the all `.bat` files with `cmd.exe`. If the suffix is `.ps1`, they would be called with `powershell` (if it's present). 

Unfortunately though, the `#!` will be treated as invalid syntax, not as a comment (which are `:: ` or `REM ` in `.bat` files)

If it's possible to detect the container/daemon OS here, I think it's fine to assume the syntax is `.bat`. If it's not possible to detect, I'd also feel ok with as special case like `shell="cmd.exe"` triggering the Windows behavior (add `.bat` suffixes, remove shebang).

Reply:
@micahyoung  If I write both `bin/build` and `bin/build.bat` will the lifecycle just pick up the right one?

Reply:
Yeah, that would work. I wrote a hybrid buildpack that does just that.

Reply:
I'll think about it. that would mean writing either an invalid `build` or `build.bat` in every case.

Reply:
I've updated this to create the hybrid buildpack (both `build` and `build.bat`)

Reply:
@jkutner I think that we would still need to remove the shebang in the windows case. https://github.com/buildpacks/pack/pull/1166/files#r634567908

Reply:
@dfreilich fixed in 81c66ff